cpp/src/rolling/rolling.cu

```diff
@@ -691,12 +712,29 @@ std::unique_ptr<column> grouped_rolling_window(table_view const& group_keys,
       return thrust::minimum<size_type>{}(following_window, (group_end - 1) - idx);
     };
 
-  return cudf::experimental::detail::rolling_window(
-    input,
-    thrust::make_transform_iterator(thrust::make_counting_iterator<size_type>(0), preceding_calculator),
-    thrust::make_transform_iterator(thrust::make_counting_iterator<size_type>(0), following_calculator),
-    min_periods, aggr, mr
+  window_wrapper grouped_preceding_window(
+    group_offsets.data().get(), 
+    group_labels.data().get(), 
+    preceding_window
+  );
+  
+  window_wrapper grouped_following_window(
+    group_offsets.data().get(), 
+    group_labels.data().get(), 
+    following_window
   );
+ 
+  if (aggr->kind == aggregation::CUDA || aggr->kind == aggregation::PTX) {
+    return cudf::experimental::detail::rolling_window_udf(input,
+                                                          grouped_preceding_window, "preceding_window_wrapper",
+                                                          grouped_following_window, "following_window_wrapper",
+                                                          min_periods, aggr, mr, 0); // XXX: What stream should be set here?
```

Comment:
The value itself should be `0`, but any functions declared in a `detail` namespace should have a `cudaStream_t stream` parameter. We don't expose a stream parameter to the user yet afaik.